src/core/logging.h

```diff
@@ -19,49 +19,34 @@
 #ifndef gpsbabel_logging_h_included
 #define gpsbabel_logging_h_included
 
-// A wrapper for QTextStream that provides a sensible Warning() and Fatal()
-// with convenient stream operators.
+// A wrapper for QDebug that provides a sensible Warning() and Fatal()
+// with convenient functions, stream operators and manipulators.
 
-#include <QtCore/QTextStream>
-#include <QtCore/QFile>
-#include <cstdlib> //exit()
+#include <QtCore/QDebug>       // for QDebug
+#include <QtCore/QFile>        // for QFile
+#include <QtCore/QIODevice>    // for QIODevice, QIODevice::WriteOnly
+#include <QtCore/QString>      // for QString
+#include <QtCore/QTextStream>  // for QTextStream
+#include <cstdio>              // for stderr
+#include <cstdlib>             // for exit
```

Comment:
What do you use for generating these clean, sorted #include trees?

Reply:
include-what-you-use.  But I almost always have to hand massage them.  The ordering conventions are different, but more importantly include-what-you-use often blows throw things like <ctime> or <cstdint> and tells you to use some underlying system dependent include file.
I also had to build custom mapping files for Qt, the defaults had a few issues.  I see they have updated their Qt imp file now, so maybe that isn't necessary.